pkg/test/devstack/default_publisher_test.go

```diff
@@ -27,46 +26,60 @@ func TestDefaultPublisherSuite(t *testing.T) {
 
 func (s *DefaultPublisherSuite) TestNoDefaultPublisher() {
 	testcase := scenario.Scenario{
-		Spec: testutils.MakeSpecWithOpts(s.T(),
-			legacy_job.WithEngineSpec(
-				model.NewWasmEngineBuilder(scenario.InlineData(cat.Program())).
-					WithEntrypoint("_start").
-					WithParameters(
-						"data/hello.txt",
-						"does/not/exist.txt",
-					).
-					Build(),
-			),
-		),
+		Job: &models.Job{
+			Name:  s.T().Name(),
+			Type:  models.JobTypeBatch,
+			Count: 1,
+			Tasks: []*models.Task{
+				{
+					Name: s.T().Name(),
+					Engine: wasmmodels.NewWasmEngineBuilder(scenario.InlineData(cat.Program())).
+						WithEntrypoint("_start").
+						WithParameters(
+							"data/hello.txt",
+							"does/not/exist.txt",
+						).
+						MustBuild(),
+				},
+			},
+		},
 		ResultsChecker: expectResultsNone,
-		JobCheckers: []legacy_job.CheckStatesFunction{
-			legacy_job.WaitForSuccessfulCompletion(),
+		JobCheckers: []scenario.StateChecks{
+			scenario.WaitForSuccessfulCompletion(),
 		},
 	}
 
 	s.RunScenario(testcase)
 }
 
 func (s *DefaultPublisherSuite) TestDefaultPublisher() {
+	// we are skipping this test because the orchestrator endpoint doesn't require a publisher
+	s.T().Skip("DefaultPublisher was a concept on the Requester endpoint, but not the orchestrator endpoint")
```

Comment:
I am not sure that makes sense. Currently we allows jobs to be submitted w/ out a publisher - since not all jobs require publishing, and for some types (daemon, service) a publisher isn't allowed. Are we sure this is a behavior we want to keep? My 2 cents would be: Remove the concept of a default publisher and leave it up to the user when they issue the job. Just seems simpler.

But for context:

https://github.com/bacalhau-project/bacalhau/pull/3430 Is the PR that added a default publisher:

This method that parses the default publisher: https://github.com/bacalhau-project/bacalhau/pull/3430/files#diff-3ba3ca7121ad93aea911fe09eb1bf3517c3fcb824079201d858f6fa49853be85R201 is wrong - `"local"` isn't supported, which is the default publisher we set in this test. 

The (old) requester API supports default publisher here: https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/requester.go#L217 and it works as expected. 

But for the (new) orchestrator API the error returned here is non-nil (says local unsupported) https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/requester.go#L235 so we never register a publisher transformer on the job when it goes through the orchestrator API. This is totally a bug.

Reply:
I understand the source of confusion coming from #3430,  but the goal of having a default publisher is to allow users to submit jobs and gets their results without having to know much about the internal details or define the s3 bucket where results will be published. 

It is true that we allow users to submit jobs without publishers, but the job will fail if they defined `ResultPath` in their job spec without specifying a publisher. So the default publisher should only kick in if no publisher was defined and a `ResultPath` was defined. Though we usually use the default publisher since we also publish stdout and stderr to the publisher, but thats something to revisit at a later time in the future.

So think of the experience in the Managed Orchestrator where we create an S3 bucket on behalf of the user, set the default publisher configuration on our orchestrator node to use that S3 bucket, and now users can submit jobs without having to specify the publisher. 

As for default publisher should only kick in for batch and ops jobs, you are right and we should have the `JobDefaults` defined per job type in https://www.notion.so/expanso/Rethinking-Configuration-435fbe87419148b4bbc5119d413786eb

For now, lets keep the behaviour as it was before

Reply:
Approving the PR, but creating a follow up issue to bring back default publisher https://github.com/bacalhau-project/bacalhau/issues/4160. Lets try to get this out before 1.4.0

Reply:
If no publisher was defined and a ResultPath was defined we won't allow the job to be submitted: https://github.com/bacalhau-project/bacalhau/blob/081eabfba0d723fbd3889d8e4e59c1ffc126ad0f/pkg/models/task.go#L140